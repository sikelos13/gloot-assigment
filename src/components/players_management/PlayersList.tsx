import React, { useState } from 'react';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import { Player } from '../../api/types/Players';
import { Button, Input } from '@material-ui/core';

interface StoresListProps {
    playersList: Player[]
    handleUpdate: (form: any) => void;
    handleDelete: (id: string) => void;
}

const PlayersList: React.FC<StoresListProps> = (({ playersList, handleDelete, handleUpdate }: StoresListProps) => {
    const [isEditMode, setMode] = useState(false);
    const [name, setName] = useState("");
    const [selectedRow, setSelectedRow] = useState("");

    const closeEditMode = () => {
        setMode(!isEditMode);
        setSelectedRow("");
    }

    const handlePrimaryAction = (isEditRow: boolean, id: string) => {
        if (isEditMode) {
            const form = {
                name,
                id
            }

            handleUpdate(form);
            closeEditMode();
        } else {
            setMode(!isEditRow);
            setSelectedRow(id);
        }
    }

    const handleSecondaryAction = (isEditRow: boolean, id: string) => {
        if (isEditRow) {
            closeEditMode();
            setName("");
        } else {
            handleDelete(id)
        }
    }

    const onChangeInput = (event: any) => {
        const value = event.target.value;
        setName(value);
    }

    return (
        <>
            {playersList.map((player: Player) => {
                const isEditRow = isEditMode && selectedRow === player.id;
                return (
                    <TableRow
                        hover
                        key={player.id}
                        className="table-row"
                    >
                        <TableCell width="40%">
                            {player.id}
                        </TableCell>
                        <TableCell width="40%">
                            {isEditMode && selectedRow === player.id
                                ? <Input
                                    defaultValue={player.name}
                                    onChange={onChangeInput}
                                />
                                : player.name
                            }
                        </TableCell>
                        <TableCell width="20%">
                            <Button
                                className="edit-button-list"
                                variant="contained"
                                color="primary"
                                size="small"
                                value={player.id}
                                onClick={() => handlePrimaryAction(isEditRow, player.id)}
                            >
                                {isEditRow ? "Save" : "Edit"}
                            </Button>
                            <Button
                                className="edit-button-list"
                                variant="contained"
                                color="secondary"
                                size="small"
                                value={player.id}
                                onClick={() => handleSecondaryAction(isEditRow, player.id)}
                            >
                                {isEditRow ? "Cancel" : "Delete"}
                            </Button>
                        </TableCell>
                    </TableRow>
                )
            })
            }
        </>
    )

});

export default PlayersList;