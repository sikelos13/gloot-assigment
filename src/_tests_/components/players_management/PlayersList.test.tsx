import PlayersList from "../../../components/players_management/PlayersList";
import * as React from 'react';
import { shallow } from 'enzyme';
import { TableCell } from "@material-ui/core";

const playersList = [
    { id: "1234", name: "check first test" },
    { id: "1235", name: "check second test" },
    { id: "1236", name: "check third test" },
    { id: "1237", name: "check fourth test" }
];
const PlayersListProps = {
    playersList: playersList,
    handleUpdate: () => {
        console.log('deleted')
    },
    handleDelete: () => {
        console.log('selected')
    }
}

describe("Players list container renders", () => {
    it('renders with props when passed in', () => {

        const result = shallow(<PlayersList {...PlayersListProps} />).contains(<TableCell />);
        expect(result).toMatchSnapshot();
    });

    it('finds the right div for mounting the players list', () => {

        const container = shallow(<PlayersList {...PlayersListProps} />);

        expect(container.find('div#table-row')).toBeTruthy();
    });
});