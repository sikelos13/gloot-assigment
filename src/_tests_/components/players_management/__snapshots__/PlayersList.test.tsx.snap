// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Players list container renders renders with props when passed in 1`] = `false`;
